frontend/src/views/submissions.js

```diff
@@ -58,6 +58,22 @@ class Submissions extends Component {
   // makes an api call to upload the selected file
   uploadData = () => {
     this.setState({ upload_status: "loading" });
+    const fileSize = this.state.selectedFile.size / 1024 / 1024; // in MiB
+    if (fileSize > 5) {
+      let alert_message =
+        "Ensure this file size is not greater than 5 MiB. Your current file size is " +
+        Number(fileSize.toPrecision(3)) +
+        " MiB.";
+      this.setState({
+        alert_message,
+        upload_status: "failure",
+      });
+      $("#submission-table-refresh-button").click();
+      setTimeout(() => {
```

Comment:
We can save some code if we don't bother with the `setTimeout` logic here: instead, just do this check before we even set the upload_status to "loading"